pkg/controller/deployment/util/deployment_util.go

```diff
@@ -816,9 +816,10 @@ func IsRollingUpdate(deployment *extensions.Deployment) bool {
 }
 
 // DeploymentComplete considers a deployment to be complete once its desired replicas equals its
```

Comment:
> Would you add a sentence that summarizes what "complete" means? Something like:

That's what the comment is about already, isn't it clear?

> We'll need to modify docs https://kubernetes.io/docs/user-guide/deployments/#complete-deployment

Yes, I will open a PR there.

> Do we expose Complete state to the users?

Not sure what you mean. We expose a Progressing=True condition with NewReplicaSetAvailable as a reason once a deployment completes.